brew install llvm --with-lldb failure

# Please always follow these steps:
- [x] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/caskroom/homebrew-cask/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed all issues and retried your prior step?
- [x] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

---

LLVM 5.0.0 installation from source (`brew install llvm --with-lldb`) failed for me. I see this in the logs:

```
[ 95%] Building CXX object tools/lldb/source/API/CMakeFiles/liblldb.dir/SBAddress.cpp.o
cd /tmp/llvm-20170908-48556-eim22i/tools/lldb/source/API && /usr/local/Homebrew/Library/Homebrew/shims/super/clang++  -DHAVE_ROUND -DLIBXML2_DEFINED -DLLDB_USE_BUILTIN_DEMANGLER -DLLVM_BUILD_GLOBAL_ISEL -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm-20170908-48556-eim22i/tools/lldb/source/API -I/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/source/API -I/tmp/llvm-20170908-48556-eim22i/tools/lldb/include -I/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include -I/tmp/llvm-20170908-48556-eim22i/include -I/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/include -I/usr/include/python2.7 -I/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/clang/include -I/tmp/llvm-20170908-48556-eim22i/tools/lldb/../clang/include -I/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/source/.  -stdlib=libc++ -fPIC -fvisibility-inlines-hidden -Werror=date-time -std=c++11 -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wstring-conversion -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.12 -fPIC   -F/System/Library/PrivateFrameworks    -o CMakeFiles/liblldb.dir/SBAddress.cpp.o -c /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/source/API/SBAddress.cpp
make -f tools/bugpoint/CMakeFiles/bugpoint.dir/build.make tools/bugpoint/CMakeFiles/bugpoint.dir/depend
cd /tmp/llvm-20170908-48556-eim22i && /usr/local/Cellar/cmake/3.9.2/bin/cmake -E cmake_depends "Unix Makefiles" /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/bugpoint /tmp/llvm-20170908-48556-eim22i /tmp/llvm-20170908-48556-eim22i/tools/bugpoint /tmp/llvm-20170908-48556-eim22i/tools/bugpoint/CMakeFiles/bugpoint.dir/DependInfo.cmake --color=
In file included from /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/source/API/SBAddress.cpp:10:
In file included from /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include/lldb/API/SBAddress.h:14:
In file included from /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include/lldb/API/SBModule.h:16:
In file included from /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include/lldb/API/SBSymbolContext.h:13:
In file included from /tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include/lldb/API/SBBlock.h:15:
/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include/lldb/API/SBTarget.h:225:41: error: no type named 'pid_t' in the global namespace; did you mean simply 'pid_t'?
                                        ::pid_t pid, // 32 bit int process ID
                                        ^~~~~~~
                                        pid_t
/tmp/llvm-20170908-48556-1sx91si/llvm-5.0.0.src/tools/lldb/include/lldb/lldb-types.h:87:18: note: 'pid_t' declared here
typedef uint64_t pid_t;
                 ^
```

Seems like this might be a missing `unistd.h` header include in the LLDB sources (?)

Gist logs: https://gist.github.com/cc8257e92390c941bbc88112ff60f3ac

--- a/Formula/llvm.rb
+++ b/Formula/llvm.rb
@@ -62,6 +62,14 @@ class Llvm < Formula
       sha256 "c0a0ca32105e9881d86b7ca886220147e686edc97fdb9f3657c6659dc6568b7d"
     end
 
+    # Fixes "error: no type named 'pid_t' in the global namespace"
+    # https://github.com/Homebrew/homebrew-core/issues/17839
+    # Already fixed in upstream trunk
+    resource "lldb-fix-build" do
+      url "https://github.com/llvm-mirror/lldb/commit/324f93b5e30.patch?full_index=1"
+      sha256 "f23fc92c2d61bf6c8bc6865994a75264fafba6ae435e4d2f4cc8327004523fb1"
+    end
+
     resource "openmp" do
       url "https://llvm.org/releases/5.0.0/openmp-5.0.0.src.tar.xz"
       sha256 "c0ef081b05e0725a04e8711d9ecea2e90d6c3fbb1622845336d3d095d0a3f7c5"
@@ -185,6 +193,12 @@ class Llvm < Formula
         pyinclude = "#{pyhome}/include/python2.7"
       end
       (buildpath/"tools/lldb").install resource("lldb")
+
+      if build.stable?
+        resource("lldb-fix-build").stage do
+          system "patch", "-p1", "-i", Pathname.pwd/"324f93b5e30.patch", "-d", buildpath/"tools/lldb"
+        end
+      end
 
       # Building lldb requires a code signing certificate.
       # The instructions provided by llvm creates this certificate in the

